app/controllers/admin/application_pets_controller.rb

```diff
@@ -0,0 +1,12 @@
+class Admin::ApplicationPetsController < ApplicationController
```

Comment:
Nice work using a joins controller! This is an "internal" path though, in that a user doesn't need to write the URL in the navbar, but a request arrives here from a button. So, there's no real need for this to have `/admin` at the beginning just cause it's not a path a user is going to. This just means that you can put this `update` in the regular ApplicationPetsController. This isn't wrong by any means but it's okay to group it with the other joins actions if you want to